hypothesis-python/src/hypothesis/internal/reflection.py

```diff
@@ -88,6 +88,12 @@ def function_digest(function):
     return hasher.digest()
 
 
+def is_typed_named_tuple(cls):
+    """Return True if cls is a subtype of `typing.NamedTuple`."""
```

Comment:
Nit: While this statement is technically true, it doesn't return `True` if and only if cls is a subtype of `typing.NamedTuple`, which makes this docstring a bit misleading.

It would maybe be helpful to have a comment as to why it's done this way rather than a subtype check of some sort? I'm sure there's a good reason, and you may even have told me at some point, but I don't have any idea what it is!